platform-sdk/docs/proposals/DP-0x/0X - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,695 @@
+5. **Native Support Lib**: Provides a set of generic functions loading native libraries in different system architectures when packaged in a jar
+   using a predefined organization so they can be accessed with JNI.
+6. **Arkworks[https://github.com/arkworks-rs]**:  arkworks is a Rust ecosystem for zkSNARK programming.
+   Our implementation uses it as the library responsible for elliptic curve cryptography.
+7. **EC-Key Utils ** is a utility module that enables the node operator to generate a bootstrapping public/private key pair.
+
+### Module organization and repositories
+![img_13.png](img_13.png)
+1. **hedera-cryptography**: This is a separate repository for hosting cryptography-related libraries.
+   It is necessary to facilitate our build process, which includes Rust libraries. It also provides independent release cycles between consensus node code and block node code.
+2. **swirlds-native-support**: Gradle module that enables loading into memory compiled native libraries so they can be used with JNI.
+3. **swirlds-cryptography-tss**: Gradle module for the TSS Library.
+   This library-only client is the consensus node so that it will remain close to it in the `hedera-services` repository under the `platform-sdk` folder.
+4. **swirlds-cryptography-signatures**: Gradle module for the Bilinear Pairings Signature Library.
+5. **swirlds-cryptography-pairings-api**: Gradle module for the Bilinear Pairings API. Minimizes the impact of adding or removing implementations.
+6. **swirlds-cryptography-alt128**: Gradle module that will implement the Bilinear Pairings API using alt-128 elliptic curve.
+   That curve has been chosen following EVM's support. The artworks rust library will provide support for it.
+   The module will include Java and Rust codes that will be compiled for all possible system architectures and distributed in a jar under a predefined structure.
+
+### Libraries Specifications
+
+#### Swirlds Crypto TSS Library
+##### Overview
+This library implements the Groth21 TSS-specific primitives.
+
+##### Constraints
+Once built, this library will depend on artifacts constructed in other repositories.
+
+##### Public API
+###### `TssService`
+**Description**: This class handles all tss specific operations. 
+###### `TssParticipantDirectory`
+**Description**: This class holds all information about the participants in the scheme. Including: participants' EC public keys, public shares, private shares, number of shares. 
+###### `TssMessage`
+**Description**: This class is used in the exchange of secret information between participants. Contains an encrypted message, a polynomial commitment, and a cryptographic proof that can be used to validate this message.
+###### `TssShareId`
+**Description**: This class represents the unique identification of a share in the system. It is linked to a FieldElement, so it can be used as input in the polynomial.
```

Comment:
"It is linked to a FieldElement"

I wouldn't say it's linked - it IS a field element

Reply:
If expressed as "it is a FieldElement" in the documentation, I would expect as a reader that it extends or implements a FieldElement interface. 
Maybe `it contains a FieldElement`? or  `it is produced out of a FieldElement`?

Reply:
Maybe "consists of a FieldElement"?